### DISCUSSION TITLE
Problems with v33.1.1.444 runtime for Windows

### DISCUSSION THREAD

**Discussion by stevenspeirs** [airsdk/Adobe-Runtime-Support#788] [Q&A]
After installing the v33.1.1.444 runtime for Windows (on Windows 10 20H2 19042.928), the installation appears to be successful ("Adobe AIR has been successfully installed.") but I cannot install any AIR apps ("The application could not be installed because the installer file is damaged. Try obtaining a new installer file from the application author.") or uninstall AIR via the Control Panel ("An error occurred while uninstalling Adobe AIR.  Uninstallation may not be allowed by your administrator. Please contact your administrator.").

I have tried to install a number of AIR apps (our own app and several sample apps which I downloaded from Adobe's web site before they closed it down) with the same results ("installer file is damaged").

Also, upgrading from v33.1.1.385 appears to be successful (""Adobe AIR has been successfully installed."") but does not seem to copy all the new files across to the C:\Program Files (x86)\Common Files\Adobe AIR\ directory (only the 'setup.msi' file is copied) and I can no longer uninstall AIR via the Control Panel.
Anyone upgrading may not notice a problem unless they check the version in place or try to uninstall.
Subsequent attempts to upgrade result in the same experience (an offer to upgrade rather than stating that the latest version is already installed, followed by the "successfully installed" message).

Manually deleting the C:\Program Files (x86)\Common Files\Adobe AIR\ directory and reverting to the v33.1.1.385 runtime works fine.

I have full administrative rights to the PC I'm using and right-clicking the AIR v33.1.1.444 runtime installer and selecting 'Run as administrator' makes no difference.
I've been using AIR since v2.0 and I've not encountered this problem before.

I don't see any other reports of such issues but users of our own app have reported similar issues since the v33.1.1.444 runtime was released so I don't think I'm alone.

Any advice would be greatly appreciated.
Thank you.

**Comment by ajwfrost** [comment]
@stevenspeirs thanks for this report. Interestingly we don't see the same issue about installing other applications, that's still working fine for us. But it does indeed appear that the .444 installer isn't properly completing hence the apps are still being installed/run as .385.

With your first description - where the AIR apps can't be installed - were you installing .444 over the top of another version? if so was it .385 or was it an Adobe deployment? I think this description is similar to what we'd heard about with some apps installing now with AIR 32 due to the expiry of a certificate...

We'll double-check on this to see what the issue is with the .444 installer; I just had a quick look myself and tried running the setup.msi which should be done as part of the normal set-up, but it gave an error about network locations... which perhaps is where the problem is.

thanks

   Andrew

**Reply by stevenspeirs** [reply]
Thanks so much for the quick reply.

My first description - where the AIR apps can't be installed and AIR cannot be uninstalled - was with a clean .444 installation (no pre-existing AIR runtime installed). All .444 files seem to be in place in the Common Files directory but it's just not possible for me to install any AIR apps or uninstall AIR itself.

After upgrading from .385 to .444, apps can still be installed as .385 is still (partially) in place (only the 'setup.msi' file is overwritten), hence the problem might not be noticed until trying to upgrade again (when the current version number is displayed as .385) or uninstall (when the error message appears).

**Reply by TEAM-OUTDOOR** [reply]
Hello,

I have exactly the same problem that you have.

First thing is that when you want to install the 33.1.1.444 aver the 33.1.1.345, it said that it is successfull, but when you check inside the control panel it is still the 33.1.1.345 installed (and the files inside the folder mentioned by @stevenspeirs are not replaced.
Then if you tried to uninstall the 33.1.1.345, you cannot, there is an error saying that you must do it with admin right, but even with admin you can't.

So I did several test, and to uninstall it, you can extract the files from the installer of the 33.1.1.345, and make uninstall from the msi that you get. Then it is possible to install the 33.1.1.444, but then if you check inside the control panel, it says that it is the 33.1.1.0 installed.
And still you get the same problem to uninstall it, solved only by using the msi file.

Next problem is that with both of these releases (33.1.1.345 and 33.1.1.444) it is impossible to install any AIR package, getting the same error as stevenspeirs : "The application could not be installed because the installer file is damaged. Try obtaining a new installer file from the application author."

So I did again some tests, and if I install the 32.0.0.125 runtime from Adobe, it is possible to install every AIR package I want.
So I think that something is really broken inside the latest 33.x releases.

Thank you for your help.

**Comment by balaedit** [comment]
Same issue with me. I disabled antivirus and it's working fine. May be antivirus blocking the .air packages.

https://github.com/Gamua/Adobe-Runtime-Support/issues/667#issuecomment-770206457

**Reply by TEAM-OUTDOOR** [reply]
I'm using only windows defender...

But still, it is not normal that we can't uninstall or update AIR.

**Reply by stevenspeirs** [reply]
@balaedit Thank you for your input. Unfortunately, like @TEAM-OUTDOOR, I am also using only Windows Defender and disabling it does not make a difference. (I experience exactly the same symptoms with Windows Defender enabled and disabled.)
Also, v33.1.1.385 works fine with Windows Defender enabled.

@ajwfrost has confirmed (above) that there appears to be a problem with the v33.1.1.444 Windows runtime installer:
"_... it does indeed appear that the .444 installer isn't properly completing..._"
"_... it gave an error about network locations..._ "

... so I think that we might just have to wait patiently until they can investigate further and resolve the issue.

**Comment by billgwx** [comment]
I've discovered the source of the problem. I deleted the C:\Program Files (x86)\Common Files\Adobe AIR\ directory as suggested above and reinstalled .444. The installation went fine, but--in the Control Panel "Programs and Features" I noticed it shows up as version 33.1.1.0.

So here's what happens when you attempt to upgrade from .385: the .444 installer, containing badly versioned files (33.1.1.0), sees .385 files already installed, rightly thinks the installed .385 files are "newer," and will not overwrite them. The .444 installer likely also leaves behind mixed versions of various files; the uninstaller, seeing these different versions, probably does not like that for security or whatever reasons, and so it fails rather than causing more trouble.

One solution would be for Harman to create a new runtime with a higher version number containing all properly versioned files (.445?), that would overwrite anything from both the .385 and .444 installers to be safe.

**Reply by ajwfrost** [reply]
Yes, thanks - we've been having fun with this today!
There were two issues, as you've said: the fact that the embedded DLL/EXE files were built with a .0 final build number means that the installers wouldn't have overwritten them, but also the package IDs that are contained within the MSI file were also meant to be updated within the ActionScript that handles installation.. so AIR treated this as a 'patch' rather than an 'upgrade' and failed to update the registry/"Add-Remove Program" settings properly. 

We have a new release available of the SDK (33.1.1.476) waiting to be published and are trying to ensure the runtime files work properly with the installation/updating of the AIR runtime .. but, for folk who had installed 33.1.1.444 over 33.1.1.385, then the best thing is to drop it back to a 'normal' 33.1.1.385 build by restoring the setup.msi file that's copied to:
`C:\Program Files (x86)\Common Files\Adobe AIR\Versions\1.0\setup.msi`

A link to setup.zip is below -> sha-256 sum = `aaed108faaa85a990e9c31fc20d2a750b56f540239054d8d004c96dd72562001`
and it just contains the .385 setup.msi -> sha-256 sum = `7194fbe6883c6d4535cf7ef80cf7cc9349523c1dbf71b84564fa702a425de216`
[setup.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/6379731/setup.zip)

Thanks
